lib/utils/compose.coffee

```diff
@@ -349,13 +349,42 @@ tagServiceImages = (docker, images, serviceImages) ->
 			logs: d.logs
 			props: d.props
 
-authorizePush = (tokenAuthEndpoint, registry, images) ->
+
+getPreviousRepos = (logger, appID) ->
+	sdk = require('balena-sdk').fromSharedOptions()
```

Comment:
Looking at the implementation of `fromSharedOptions()`, it seems a non-trivial amount setting up, unclear if actually establishing network connections. It might be better to reuse the same sdk object between `getPreviousRepos` and `authorizePush`, but moving its creation to the common parent function (`deployProject`) and passing it down as argument.